src/main/resources/XWiki/Macros/Team.xml

```diff
@@ -0,0 +1,1407 @@
+        <separators>|</separators>
+        <size>1</size>
+        <unmodifiable>0</unmodifiable>
+        <values>Current User|Current Wiki|Global</values>
+        <classType>com.xpn.xwiki.objects.classes.StaticListClass</classType>
+      </visibility>
+    </class>
+    <property>
+      <async_cached>0</async_cached>
+    </property>
+    <property>
+      <async_context/>
+    </property>
+    <property>
+      <async_enabled>0</async_enabled>
+    </property>
+    <property>
+      <code>{{velocity}}
+#set($discard = $xwiki.ssx.use('xwiki:XWiki.Macros.Team'))
+#set($discard = $xwiki.jsx.use('xwiki:XWiki.Macros.Team'))
+#set($pictureList = {})
+#macro(avatarInitials $name $size $letterAvatarBgColor $letterAvatarFontColor)
+  &lt;span class="xwikiteam-avatar xwikiteam-avatar-initials" style="height:${size}px; width:${size}px; border-radius: ${size}px; background-color: $letterAvatarBgColor"&gt;
+    &lt;span class="xwikiteam-avatar-initials-inner"&gt;
+      &lt;span class="xwikiteam-avatar-initials-letters" style="color: $letterAvatarFontColor"&gt;
+        #foreach($firstLetter in $name.toUpperCase().split("[\s-]"))$firstLetter.charAt(0)#end
+      &lt;/span&gt;
+    &lt;/span&gt;
+  &lt;/span&gt;
+#end
+#macro(avatar $username $size $disableLetterAvatars $letterAvatarBgColor $letterAvatarFontColor)
+  #set($displayUser = $xwiki.getUserName($username, false))
+  #set($fontSize = $size / 2.5)
+  &lt;li class="xwikiteam-user" data-name-without-accents="${util.clearAccents($displayUser.toLowerCase())}" title="$displayUser" data-username="$username" style="font-size: ${fontSize}px"&gt;
+    #set($url = $xwiki.getURL($username))
+    #set($pictureListKey = "${size}-${username}")
+    #if(!$pictureList.containsKey($pictureListKey))
+      #set($pictureURL = '')
+      #set($profileDoc = $xwiki.getDocument($username))
+      #if(!$profileDoc.isNew())
+        #if($profileDoc.attachmentList.size() != 0)
+          #set($profileobj = $profileDoc.getObject('XWiki.XWikiUsers', 0))
+          #set($profileObjAvatar = $profileDoc.getValue('avatar'))
+          #if($profileObjAvatar &amp;&amp; $profileObjAvatar != '')
+            #foreach($attach in $profileDoc.attachmentList)
+              #if($attach.filename == $profileDoc.getValue('avatar'))
+                #set($pictureURL = $profileDoc.getAttachmentURL($profileObjAvatar, 'download', "height=$size"))
+                #break
+              #end
+            #end
+          #end
+        #end
+      #end
+      #if($pictureURL == "" &amp;&amp; !$disableLetterAvatars)
+        #set($picture = "#avatarInitials($displayUser $size $letterAvatarBgColor $letterAvatarFontColor)")
+      #else
+        #if($pictureURL == "")
+          #set($pictureURL = $xwiki.getAttachmentURL('XWiki.XWikiUserSheet', 'noavatar.png'))
+        #end
+        #set($picture = "&lt;img class='xwikiteam-avatar' src='$pictureURL' alt='$displayUser' title='$displayUser' width='$size' height='$size' /&gt;")
```

Comment:
Missing XML escaping.

Reply:
Should  $pictureURL be escaped? (I realize that the other values need to be escaped)

Reply:
Yes, definitely. From a short look this may be a URL that may contain several parameters which are separated by `&` which needs to be escaped  - we try to produce valid XML and not rely on the fact that these are most likely not [ambiguous ampersands](https://mathiasbynens.be/notes/ambiguous-ampersands). There might also be other reasons (user-supplied attachment names we cannot trust, ...) but the fact that this might be a URL with parameters is enough.

Reply:
ok, thanks!